### PR TITLE
Flatten typings, d.ts rollup

### PR DESCRIPTION
**Pull request by dherges**
## I'm submitting a...

```
[ ] Bug Fix
[x] Feature
[ ] Other (Refactoring, Added tests, Documentation, ...)
```

## Checklist

- [ ] Commit Messages follow the [Conventional Commits](https://conventionalcommits.org/) pattern
  - A feature commit message is prefixed "feat:"
  - A bugfix commit message is prefixed "fix:"
- [ ] Tests for the changes have been added

## Description

See #1175 

See angular/angular#29086

## Does this PR introduce a breaking change?

```
[ ] Yes
[x] No
```

### CODE REVIEW

**Comment by dherges** [comment]
Findings:

- for human beings, it would be best to publish the public `.d.ts` from the entry point only
  - does not contain old greeks
  - does not contain `@internal` annotated symbols
- for the compiler, i think we need to publish the internal `.d.ts` from the generated bundle index
  - contains old greeks
  - contains `@internal` symbols

Further tasks:
 - remove single module `.d.ts` files from dest folder
 - store typings in `<dest>/typings` folder for all entry points (similar to `bundles`, `fesm5` folder, and so on)

Nice-To-Have:
 - is it possible to re-use tsc compiler program for extracting API?

File: package.json [hunk 1]
```diff
@@ -27,6 +27,7 @@
   "main": "./public_api.js",
   "typings": "./public_api.d.ts",
   "dependencies": {
+    "@microsoft/api-extractor": "^6.3.0",
```
Comment (alan-agius4) [inline package.json]: Ideally version 7 is used. Version 6 can be challenging to use in existing projects.

**Comment by alan-agius4** [comment]
You also need to re-wire the ngc metadata.

See https://github.com/angular/angular/pull/28588 for more context

**Comment by alan-agius4** [comment]
Running this in watch mode might make incremental builds significantly slower since it cannot used a previous program not the file cache.

The main reason why is the api extractor doesn’t nesserary support the latest or same ts version that the uses have installed.

**Pull request closed without merging**